src/main/java/unicash/ui/StyleSheet.java

```diff
@@ -1,23 +1,170 @@
     public static final String TEXT_FILL_GREEN = "-fx-text-fill: green";
     public static final String TEXT_FILL_BLACK = "-fx-text-fill: black";
     public static final String TEXT_FILL_WHITE = "-fx-text-fill: white";
 
-    // Font styling
+    /* Font styling */
     public static final String FONT_STYLE_BOLD = "-fx-font-weight: bold";
     public static final String FONT_STYLE_REGULAR = "-fx-font-weight: normal";
     public static final String FONT_STYLE_ITALIC = "-fx-font-style: italic";
 
-    // Custom text formatting
+    /* Custom text formatting */
     public static final String TRANSACTION_ID_SEPARATOR = ". ";
+    public static final String HEX_STRING_PADDING_SPECIFIER = "%6s";
+    public static final String HEX_COLOR_STRING_SPECIFIER = "%02X%02X%02X";
+
+    /* Background styling and formatting */
+    public static final String TEXT_BACKGROUND_COLOR = "-fx-background-color: %s";
+
+    /* Color utilities */
+    public static final int MAX_COLOUR_VALUE = 255;
+    public static final int MEDIAN_COLOR_VALUE = 128;
+    public static final int HEXADECIMAL_RADIX = 16;
+    public static final String HEXADECIMAL_RADIX_SYMBOL = "#";
+    public static final int COLOR_START_INDEX = 0;
+    public static final int COLOR_END_INDEX = 6;
+
+    /* Color variables and offsets to be modified according to preference */
+    public static final int BRIGHTNESS_THRESHOLD = 130; // Higher threshold = brighter value, 130 - 140 ideal
+    public static final int BRIGHTNESS_OFFSET = 60; // Default is 64
+    public static final int RED_OFFSET = 90;
+    public static final int GREEN_OFFSET = 90;
+    public static final int BLUE_OFFSET = 0;
+    public static final boolean IS_YELLOW_SKEW = true;
+
+    /**
+     * Returns a 6-digit hexadecimal number based on the Category's unique
+     * hash code, to be used as a color indicator, without any color offsetting.
+     */
+    public static String getCategoryColorFromHash(Object obj) {
+
+        // Absolute value of hash code taken to guard against negative hash values
+        String hexString = Integer.toHexString(Math.abs(obj.hashCode()));
+
+        // Padding for hash codes that are less than 6 digits long
+        hexString = padHexString(hexString);
+
+        // Hex index refers to the specific substring of the hash code to be used.
+        // This can be altered according to specific hashing patterns.
+        String categoryColorString = hexString.substring(COLOR_START_INDEX,
+                COLOR_END_INDEX).toUpperCase();
+
+        return getColorOutput(categoryColorString);
+    }
+
+    /**
+     * Returns a 6-digit hexadecimal number based on an Object's unique
+     * hash code, to be used as a color indicator. If the hex color value
+     * generated from the hashcode accords poor {@code Label} visibility,
+     * the value is offset such that is always of a certain brightness value.
+     */
+    public static String getBrightCategoryColorFromHash(Object obj) {
```

Comment:
If we're using this in the constraints of a `String`, can we make the type parameter a `String` instead? I don't personally see a compelling reason for `Object` to be used since it leaves potentially "undesired" behavior as we perform unit testing/during PE testing, like having to account for different input types

Reply:
This occured to me while speaking with @elhy1999 about his Statistics representation, I realized that this process of getting a color from an `Object` can be applied universally across the application, whereever there is a UI element with colors involved, this can be used to generate a color specific to any element. Given that this method also has built in color correction processes, all UI elements (or really, anything with a hash code, which is everything) involving color would have a consistent color scheme. 

Ngl I honestly didn't really think of this when writing the method initially but now I think it's a pretty neat way to ensure color consistency across our App.

Reply:
If that is the case, then could we consider using using generics and method overloading to constraint the method signatures? I agree it's a neat way to ensure color consistency but I am against leaving the parameters as just a generic `Object`. 

Since we're applying this to any UI elements, it could be a generic that extends the `UiPart` type or `Category` type and both lead to a common method that takes in an `int hash` instead of a pure `Object`. 

From the perspective of testing, an `Object` parameters requires way more extensive testing as it can be anything that goes in, this comment is more so to raise the potential code design flaw of just working with pure Objects when we can perhaps constraint it further using generics and avoid any foot guns with using Objects.

Reply:
Sure, that makes sense, I'll commit the changes and modify the StyleSheet unit tests as well.